template/cordova/lib/deployment.js

```diff
@@ -0,0 +1,275 @@
+
+        var devices = matchedLines.map(function(line, arrayIndex) {
+            var match = line.match(LINE_TEST);
+            var index = parseInt(match[1], 10);
+            var name = match[2];
+
+            var shorthand = '';
+            var type = 'emulator';
+
+            if (name === 'Device') {
+                shorthand = 'de';
+                type = 'device';
+            } else if (arrayIndex === 1) {
+                shorthand = 'xd';
+            } else {
+                shorthand = index;
+            }
+            var deviceInfo = new DeviceInfo(index, name, type);
+            deviceInfo.__sourceLine = line;
+            deviceInfo.__shorthand = shorthand;
+            return deviceInfo;
+        });
+
+        return devices;
+    });
+};
+
+AppDeployCmdTool.prototype.installAppPackage = function(pathToAppxPackage, targetDevice, shouldLaunch, shouldUpdate, pin) {
+    var command = shouldUpdate ? '/update' : '/install';
+    if (shouldLaunch) {
+        command += 'launch';
+    }
+
+    return run(this.path, [command, pathToAppxPackage, '/targetdevice:' + targetDevice.__shorthand]);
+};
+
+AppDeployCmdTool.prototype.uninstallAppPackage = function(packageInfo, targetDevice) {
+    return run(this.path, ['/uninstall', packageInfo, '/targetdevice:' + targetDevice.__shorthand]);
+};
+
+AppDeployCmdTool.prototype.launchApp = function(packageInfo, targetDevice) {
+    return run(this.path, ['/launch', packageInfo, '/targetdevice:' + targetDevice.__shorthand]);
+};
+
+function WinAppDeployCmdTool(targetOsVersion) {
+    if (!(this instanceof WinAppDeployCmdTool))
+        throw new ReferenceError('Only create a WinAppDeployCmdTool as an instance object.');
+
+    DeploymentTool.call(this);
+    this.targetOsVersion = targetOsVersion;
+    /* jshint ignore:start */ /* Ignore jshint to use dot notation for 2nd process.env access for consistency */
+    var programFilesPath = process.env['ProgramFiles(x86)'] || process.env['ProgramFiles'];
+    this.path = path.join(programFilesPath, 'Windows Kits', '10', 'bin', 'x86', 'WinAppDeployCmd.exe');
+    /* jshint ignore:end */
+}
+
+WinAppDeployCmdTool.prototype = Object.create(WinAppDeployCmdTool);
+WinAppDeployCmdTool.prototype.constructor = WinAppDeployCmdTool;
+
+WinAppDeployCmdTool.prototype.enumerateDevices = function() {
+    var that = this;
+    var LINE_TEST = /^([\d\.]+?)\s+([\da-fA-F\-]+?)\s+(.+)$/m;
+
+    return run(that.path, ['devices']).then(function(result) {
+        var lines = result.split('\n');
+        var matchedLines = lines.filter(function(line) {
+            return LINE_TEST.test(line);
+        });
+
+        var devices = matchedLines.map(function(line, arrayIndex) {
+            var match = line.match(LINE_TEST);
+            var ip = match[1];
+            var guid = match[2];
+            var name = match[3];
+            var type = 'device';
+
+            var deviceInfo = new DeviceInfo(arrayIndex, name, type);
+            deviceInfo.__ip = ip;
```

Comment:
It's "private" data - in other words, DeviceInfo promises an index, name, and type property.  The other properties that hang off of them are specific to the thing that created them.  Since we don't have Symbols yet, these will have to do.